src/LsmTree.hs

```diff
@@ -13,18 +13,27 @@ module LsmTree
        , initFileSystem
        , insertTree
        , insertTable
+         -- * foldr
+       , sizeTree
        ) where
 
 -- | Data type that stores values while program is working.
 data Tree a
-    = TreeNode (Tree a) a (Tree a)
+    = TreeNode (Tree a) !a (Tree a)
     | Leaf
     deriving stock (Eq, Show)
 
+instance Foldable Tree where
+    foldMap _ Leaf = mempty
+    foldMap f (TreeNode l x r) = foldMap f l <> f x <> foldMap f r
```

Comment:
It's better to implement `foldr` for `Tree` as well, since it will be more efficient.

Reply:
how I need to traverse through the tree. Inorder or preorder? Or maybe levelorder, what is better?

Reply:
In depends on your desired behavior. But you should keep in mind that function `toList` is implemented like this:

```haskell
toList = foldr (:) []
```

So, if you want to convert your tree to a sorted list, it probably should be inorder.